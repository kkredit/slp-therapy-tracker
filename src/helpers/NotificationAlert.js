import React from 'react';
import Alert from 'react-bootstrap/Alert'

import './NotificationAlert.css';

function NotificationAlert (props) {
  if (!props.active) {
    return null;
  } else {
    return (
      <Alert className="notification" key={props.key} variant={props.variant}>
        {props.text}
      </Alert>
    );
  }
};

export default NotificationAlert;
