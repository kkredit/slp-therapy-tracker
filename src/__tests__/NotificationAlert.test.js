import React from 'react';
import ReactDOM from 'react-dom';
import NotificationAlert from '../helpers/NotificationAlert.js';
import { create } from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NotificationAlert />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Snapshots',()=>{
  test('No alert is default', () => {
    let tree = create(<NotificationAlert />);
    expect(tree.toJSON()).toMatchSnapshot();
  })

  test('Alert turns on', () => {
    let tree = create(<NotificationAlert active={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
