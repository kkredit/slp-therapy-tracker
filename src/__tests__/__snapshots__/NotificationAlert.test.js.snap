// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Alert turns on 1`] = `
<div
  className="fade notification alert show"
  role="alert"
/>
`;

exports[`Snapshots No alert is default 1`] = `null`;
